third_party/move/move-model/src/lib.rs

```diff
@@ -61,10 +61,49 @@ pub mod ty;
 pub mod well_known;
 
 // =================================================================================================
-// Entry Point
+// Entry Point V2
+
+/// Represents information about a package: the sources it contains and the package private
+/// address mapping.
+#[derive(Debug, Clone)]
+pub struct PackageInfo {
+    pub sources: Vec<String>,
+    pub address_map: BTreeMap<String, NumericalAddress>,
+}
 
-/// Build the move model with default compilation flags and default options and no named addresses.
-/// This collects transitive dependencies for move sources from the provided directory list.
+/// Builds the Move model for the v2 compiler. This builds the model, compiling both code
+/// and specs from sources into the typed check AST. No bytecode is attached to the model.
+/// This currently uses the v1 compiler as the parser (up to expansion AST), after that
+/// a new type checker.
+pub fn run_model_builder_in_compiler_mode(
+    source: PackageInfo,
+    deps: Vec<PackageInfo>,
+) -> anyhow::Result<GlobalEnv> {
+    let to_package_paths = |PackageInfo {
+                                sources,
+                                address_map,
+                            }| PackagePaths {
+        name: None,
+        paths: sources,
+        named_address_map: address_map,
+    };
+    run_model_builder_with_options_and_compilation_flags(
+        vec![to_package_paths(source)],
+        deps.into_iter().map(to_package_paths).collect(),
+        ModelBuilderOptions {
+            compile_via_model: true,
+            ..ModelBuilderOptions::default()
+        },
+        Flags::model_compilation(),
+    )
+}
+
+// =================================================================================================
+// Entry Point V1
+
+/// Build the move model with default compilation flags and default options. This calls
+/// the move compiler v1 to compiler to bytecode first and attach the generated bytecode to
```

Comment:
nit: compiler --> compile

Reply:
This is not added code but rather displaced by the added "Entry Point V2" code, so perhaps it should be left alone.  But if Wolfgang wants to correct it, then also fix "Build the move model" --> "Build**s** the move model..."

Reply:
You see the need for this when you look at the entire file. Without this, its rather difficult to navigate. The Diff alone does not show you this.